src/ralph/backends/data/clickhouse.py

```diff
@@ -0,0 +1,598 @@
+
+        return inserted_count
+
+    def bulk_delete(
+        self, batch: list, ignore_errors: bool = False, event_table_name=None
+    ):
+        """Deletes a batch of documents from the selected database table."""
+        try:
+            if event_table_name is None:
+                event_table_name = self.event_table_name
+            self.client.command(
+                "ALTER TABLE {event_table_name} DELETE WHERE event_id IN {batch}"
+            )
+        except (ClickHouseError, BackendException) as error:
+            if not ignore_errors:
+                raise BackendException(*error.args) from error
+            logger.warning(
+                "Bulk delete failed for current chunk but you choose to ignore it.",
+            )
+            # There is no current way of knowing how many rows from the batch
+            # succeeded, we assume 0 here.
+            return 0
+
+        deleted_count = len(batch)
+        logger.debug("Deleted %d documents chunk with success", deleted_count)
+
+        return deleted_count
+
+    def bulk_update(
+        self, batch: list, ignore_errors: bool = False, event_table_name=None
+    ):
+        """Update a batch of documents into the selected database table."""
+        try:
+            if event_table_name is None:
+                event_table_name = self.event_table_name
+
+            #sql = f"ALTER TABLE {event_table_name} UPDATE event = event_str "
+            #"WHERE event_id IN {batch}"
+            # self.client.command(sql)
+        except (ClickHouseError, BackendException) as error:
+            if not ignore_errors:
+                raise BackendException(*error.args) from error
+            logger.warning(
+                "Bulk update failed for current chunk but you choose to ignore it.",
+            )
+            # There is no current way of knowing how many rows from the batch
+            # succeeded, we assume 0 here.
+            return 0
+
+        modified_count = len(batch)
+        logger.debug("Updated %d documents chunk with success", modified_count)
+        return modified_count
+
+    def write(
+        self,
+        data: Union[IOBase, Iterable[Union[bytes, dict]]],
+        target: Union[None, str] = None,
+        chunk_size: Union[None, int] = None,
+        ignore_errors: bool = False,
+        operation_type: Union[None, BaseOperationType] = None,
+    ) -> int:
+        """Writes documents from the `stream` to the instance table.
+
+        Args:
+            data: The data to write to the database.
+            target: The target table to write to.
+            chunk_size: The number of documents to write at once.
+            ignore_errors: Whether to ignore errors or not.
+            operation_type: The operation type to use for the write operation.
+        """
+
+        event_table_name = self.event_table_name
+        if target:
+            event_table_name = target
```

Comment:
Why do you need another variable `event_table_name` instead of using `target` directly?